R/driver-sql-server.R

```diff
@@ -107,6 +107,30 @@ setMethod("dbExistsTable", c("Microsoft SQL Server", "SQL"),
   }
 )
 
+#' @details Call catalog-specific sp_tables to make sure we list the schemas in the
+#' appropriate database/catalog.
+#' @rdname SQLServer
+#' @usage NULL
+setMethod(
+  "odbcConnectionSchemas", "Microsoft SQL Server",
+  function(conn, catalog_name = NULL) {
+
+    if (is.null(catalog_name) || !nchar(catalog_name)) {
+      return(callNextMethod())
+    }
+    sproc <- paste(
+      dbQuoteIdentifier(conn, catalog_name),
+      dbQuoteIdentifier(conn, "dbo"),
+      dbQuoteIdentifier(conn, "sp_tables"), sep = ".")
+
+    dbGetQuery(conn, paste0("
+               EXEC ", sproc, "
+               @table_name = '',
+               @table_owner = '%',
+               @table_qualifier = ''"))$TABLE_OWNER
```

Comment:
```suggestion
res <- dbGetQuery(conn, paste0(
    "EXEC ", sproc,
    "@table_name = '', ",
    "@table_owner = '%', ",
    "@table_qualifier = ''", 
  ))
  res$TABLE_OWNER
```

I think that's a bit more inline with our usual style.